frontend/integration-tests/tests/kubevirt/non-admin.scenario.ts

```diff
@@ -0,0 +1,140 @@
+/* eslint-disable no-undef */
+import { $, $$, browser, ExpectedConditions as until } from 'protractor';
+import { appHost, testName } from '../../protractor.conf';
+import { nonAdminSecret, nonAdminProvider } from './utils/mocks';
+import { click, createResources } from './utils/utils';
+import { KUBEADMIN_IDP, KUBEADMIN_USERNAME, BRIDGE_KUBEADMIN_PASSWORD, BRIDGE_HTPASSWD_IDP, BRIDGE_HTPASSWD_USERNAME, BRIDGE_HTPASSWD_PASSWORD } from './utils/consts';
+import * as loginView from '../../views/login.view';
+import * as crudView from '../../views/crud.view';
+import * as yamlView from '../../views/yaml.view';
+import { createWithWizardLink, createWithYAMLLink, closeWizard } from '../../views/kubevirt/wizard.view';
+import { logout } from '../../views/kubevirt/login.view';
+import { execSync } from 'child_process';
+
+describe('Test nonadmin user behaviour', () => {
+  const nonAdminNS = `${testName}-nonadmin`;
+  const userRole = ['view', 'edit'];
+
+  const verifyPermissions = async function( perm: string, ns: string) {
+    await browser.get(`${appHost}/k8s/ns/${ns}/virtualmachines`);
+    if (perm === 'noAccess') {
+      // crudView.isLoaded doesn't work here.
+      await browser.sleep(6000);
+      expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'view') {
+      await crudView.isLoaded(); // no error indicates it can view resource.
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'edit') {
+      await crudView.isLoaded();
+      await click(crudView.createItemButton);
+      await click(createWithWizardLink);
+      await crudView.isLoaded();
+      await click(closeWizard);
+
+      await click(crudView.createItemButton);
+      await click(createWithYAMLLink);
+      await yamlView.isLoaded();
+    }
+
+    await browser.get(`${appHost}/k8s/ns/${ns}/vmtemplates`);
+    if (perm === 'noAccess') {
+      await browser.sleep(6000);
+      expect(crudView.errorMessage.getText()).toContain('cannot list resource');
+    }
+    if (perm === 'view') {
+      await crudView.isLoaded(); // no error indicates it can view resource.
+      // TODO: check it can't use vm dialog once BZ1728523 is fixed.
+    }
+    if (perm === 'edit') {
+      await crudView.isLoaded();
+      await click(crudView.createItemButton);
+      await click(createWithWizardLink);
+      await crudView.isLoaded();
+      await click(closeWizard);
```

Comment:
shouldn't we also check the yaml in VM Templates?

Reply:
The VM and templates are very similar. Can you please make it generic?

Reply:
it's checking whether user has permission to use wizard dialog and yaml dialog, VM has 'create from yaml' dialog but VM template doesn't have it.

Reply:
VM Template newly received this functionality in  https://github.com/openshift/console